libvast/src/system/disk_monitor.cpp

```diff
@@ -101,20 +99,27 @@ disk_monitor(disk_monitor_actor::stateful_pointer<disk_monitor_state> self,
                                            self->state.dbdir, err));
       // TODO(ch20006): Add some check on the overall structure on the db dir.
       std::vector<partition_diskstate> partitions;
-      for (const auto& file : index_dir) {
-        auto partition = file.path().stem().string();
+      for (const auto& entry : index_dir) {
+        auto partition = entry.path().stem().string();
         if (partition == "index.bin")
           continue;
         uuid id;
         if (!parsers::uuid(partition, id)) {
           VAST_VERBOSE("{} failed to find partition {}", self, partition);
           continue;
         }
-        // TODO: Wrap a more generic `stat()` using `vast::path`.
-        struct stat statbuf;
-        if (::stat(file.path().c_str(), &statbuf) < 0)
-          continue;
-        partitions.push_back({id, statbuf.st_size, statbuf.st_mtime});
+
+        if (entry.is_regular_file()) {
+          std::error_code err{};
+          const auto file_size = entry.file_size(err);
+          const auto mtime = entry.last_write_time(err);
+          if (!err && file_size != static_cast<std::uintmax_t>(-1))
```

Comment:
I [tried to find](https://en.cppreference.com/w/cpp/filesystem/directory_entry/last_write_time) when `last_write_time` can return a value with all bits set to 1, and why it's an error. At least high-level docs don't hint at an issue. Mind sharing you insight on this check?

Reply:
The first part of the `if` check here is to catch the error case for `last_write_time` per https://en.cppreference.com/w/cpp/filesystem/directory_entry/last_write_time where it states:

> The overload taking a std::error_code& parameter sets it to the OS API error code if an OS API call fails

The second part of the `if` check here is to catch the `file_size` error case per https://en.cppreference.com/w/cpp/filesystem/file_size where it states:

> The non-throwing overload returns static_cast<std::uintmax_t>(-1) on errors.

Does that answer your question? I think you may have misread the code as the check isn't for `last_write_time` having all its bits set.

Reply:
Thanks for providing the references.

> I think you may have misread the code as the check isn't for `last_write_time` having all its bits set.

Yes, my knee-jerk reaction to seeing `static_cast<size_t>(-1)` (or for other unsigned arithmetic types) is that someone uses a lazy way to get the largest possible value, instead of using `numeric_limits<size_t>::max()`.